gems/smithy-client/lib/smithy-client/cbor_codec.rb

```diff
@@ -0,0 +1,112 @@
+# frozen_string_literal: true
+
+require 'base64'
+
+module Smithy
+  module Client
+    # Codec that serializes and deserializes in CBOR format.
+    class CBORCodec
+      include Shapes
+
+      # @option options [Boolean] :xml_flattened (nil)
+      def initialize(options = {})
+        # TODO: Need to add implementation to handle this trait
+        @xml_flattened = options[:xml_flattened] || false
```

Comment:
I'd leave the `TODO` in place, but I think remove the actual option here..... I think we also would want the name to be some thing other than xml_flattened.   Either `query_compatible` or maybe more specifically on the behavior, which is `default_empty_flattened` or something like that.